### PR TITLE
D3 hydrograph

### PR DESCRIPTION
**Pull request by skaymen**
Created a hydrograph using D3 according to the specifications in FV-13. Made changes to hydrograph utils and unit tests accordingly. Removed NVD3 from code and switched to latest version of D3.

### CODE REVIEW

File: floodviz/templates/index.html [hunk 1]
```diff
@@ -3,8 +3,13 @@
 
 <head>
 	<meta charset="UTF-8">
+<<<<<<< HEAD
```
Comment (skaymen) [inline floodviz/templates/index.html]: Fixed.

File: floodviz/tests/hydrograph_utils_tests.py [hunk 2]
```diff
@@ -18,14 +18,14 @@ def setUp(self):
     [{"value": [{"value": "446", "qualifiers": ["A"], "dateTime": "2008-05-20T00:00:00.000-05:00"}]}]}}}
 
     ## Returned value is different than the mocked request response because req_hydrodata accessess two extra levels of the dict before returning.
-    self.valid_return = {"sourceInfo": {"siteName": "Black Hawk Creek at Hudson, IA", "siteCode": 
-    [{"value": "05463500", "network": "NWIS", "agencyCode": "USGS"}], "timeZoneInfo": {"defaultTimeZone": 
+    self.valid_return = {"sourceInfo": {"siteName": "Black Hawk Creek at Hudson, IA", "siteCode":
```
Comment (skaymen) [inline floodviz/tests/hydrograph_utils_tests.py]: Fixed.

File: floodviz/static/js/hydrograph.js [hunk 3]
```diff
@@ -1,49 +1,66 @@
...
+        .attr("width", width + margin.left + margin.right)
+        .attr("height", height + margin.top + margin.bottom)
+        .append("g")
+        .attr("transform",
+              "translate(" + margin.left + "," + margin.top + ")");
+
+// Get the data
+d3.json("../static/data/hydrograph_data.json", function(error, data) {
+    data.forEach(function(d) {
+		d.time_mili = d.time_mili;
```
Comment (skaymen) [inline floodviz/static/js/hydrograph.js]: Removed.

File: floodviz/static/js/hydrograph.js [hunk 4]
```diff
@@ -1,49 +1,66 @@
...
+		d.value = +d.value;
```
Comment (skaymen) [inline floodviz/static/js/hydrograph.js]: Changed.

File: floodviz/hydrograph_utils.py [hunk 5]
```diff
@@ -6,44 +6,38 @@ def parse_hydrodata(jdata):
...
-            all_series_data.append({'key': site_name, 'values': []})
+            site_name = site['sourceInfo']['siteName']
+            site_id = site['sourceInfo']['siteCode'][0]['value']
 
-            # Fill new data for this series
-            for idx2, obj in enumerate(site['values'][0]['value']):
+            # Fill data for this series
+            for obj in site['values'][0]['value']:
                 value = obj['value']
                 dt = obj['dateTime']
```
Comment (mbucknell) [inline floodviz/hydrograph_utils.py]: In the future, we should consider a utility function that takes an NWIS data string and returns a python datetime object. No changes needed right now.

File: floodviz/static/css/style.css [hunk 6]
```diff
@@ -1,6 +1,22 @@
 html, body, .main-content {
-	height: 100%;
-	margin: 0;
+    height: 100%;
```
Comment (mbucknell) [inline floodviz/static/css/style.css]: Should use tabs in css files. Looks like you are using spaces.

File: floodviz/static/js/hydrograph.js [hunk 7]
```diff
@@ -1,49 +1,62 @@
...
-			d3.select('#hydrograph svg').datum(data).call(chart);
-
-			nv.utils.windowResize(chart.update);
-			return chart;
-	
-		// });
-	
-	});
+// Set the dimensions of the canvas / graph
+var margin = {top: 30, right: 20, bottom: 30, left: 50},
```
Comment (mbucknell) [inline floodviz/static/js/hydrograph.js]: Each variable should be declared as var, e.g. var width and var height. It makes it easier to see what is a variable

File: floodviz/static/js/hydrograph.js [hunk 8]
```diff
@@ -1,49 +1,62 @@
...
+    width = 600 - margin.left - margin.right,
```
Comment (mbucknell) [inline floodviz/static/js/hydrograph.js]: Where does 600 and 270 come from? Is the width and height of the visualization supposed to come from configuration?

File: floodviz/tests/hydrograph_utils_tests.py [hunk 9]
```diff
@@ -4,103 +4,110 @@
...
-
-  def test_bad_url(self):
-    self.assertEqual(req_hydrodata(self.sites, self.H_START_DT, self.H_END_DT, 'badurl.noob' ), None)
-    self.assertEqual(req_hydrodata(self.sites, self.H_START_DT, self.H_END_DT, ''), None)
- 
+    def setUp(self):
+
+        self.mock_response = {"value": {"timeSeries": {"sourceInfo": {"siteName": "Black Hawk Creek at Hudson, IA", "siteCode":
+            [{"value": "05463500", "network": "NWIS", "agencyCode": "USGS"}], "timeZoneInfo": {"defaultTimeZone":
+                                                                                                   {"zoneOffset": "-06:00", "zoneAbbreviation": "CST"}, "daylightSavingsTimeZone": {"zoneOffset": "-05:00", "zoneAbbreviation": "CDT"},
```
Comment (mbucknell) [inline floodviz/tests/hydrograph_utils_tests.py]: At some point this should be formatted so it's a little easier to see. No need to change now.

File: floodviz/hydrograph_utils.py (hunk 5 above)
Comment (skaymen) [inline floodviz/hydrograph_utils.py]: Ok.

File: floodviz/static/css/style.css (hunk 6 above)
Comment (skaymen) [inline floodviz/static/css/style.css]: Fixed.

File: floodviz/static/js/hydrograph.js (hunk 7 above)
Comment (skaymen) [inline floodviz/static/js/hydrograph.js]: Fixed.

File: floodviz/static/js/hydrograph.js (hunk 8 above)
Comment (skaymen) [inline floodviz/static/js/hydrograph.js]: Fixed.

**Pull request merged**